Python/pywarpx/_libwarpx.py

```diff
@@ -30,112 +30,294 @@
+        self.libwarpx.warpx_getF_pml_nodal_flag.restype = _LP_c_int
+        self.libwarpx.warpx_getG_pml_nodal_flag.restype = _LP_c_int
+
+        #self.libwarpx.warpx_getPMLSigma.restype = _LP_c_real
+        #self.libwarpx.warpx_getPMLSigmaStar.restype = _LP_c_real
+        #self.libwarpx.warpx_ComputePMLFactors.argtypes = (ctypes.c_int, c_real)
+        self.libwarpx.warpx_addNParticles.argtypes = (ctypes.c_char_p, ctypes.c_int,
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      ctypes.c_int,
+                                                      _ndpointer(c_particlereal, flags="C_CONTIGUOUS"),
+                                                      ctypes.c_int)
+
+        self.libwarpx.warpx_getProbLo.restype = c_real
+        self.libwarpx.warpx_getProbHi.restype = c_real
+        self.libwarpx.warpx_getCellSize.restype = c_real
+        self.libwarpx.warpx_getistep.restype = ctypes.c_int
+        self.libwarpx.warpx_gett_new.restype = c_real
+        self.libwarpx.warpx_getdt.restype = c_real
+        self.libwarpx.warpx_maxStep.restype = ctypes.c_int
+        self.libwarpx.warpx_stopTime.restype = c_real
+        self.libwarpx.warpx_finestLevel.restype = ctypes.c_int
+        self.libwarpx.warpx_getMyProc.restype = ctypes.c_int
+        self.libwarpx.warpx_getNProcs.restype = ctypes.c_int
+
+        self.libwarpx.warpx_EvolveE.argtypes = [c_real]
+        self.libwarpx.warpx_EvolveB.argtypes = [c_real]
+        self.libwarpx.warpx_FillBoundaryE.argtypes = []
+        self.libwarpx.warpx_FillBoundaryB.argtypes = []
+        self.libwarpx.warpx_UpdateAuxilaryData.argtypes = []
+        self.libwarpx.warpx_SyncCurrent.argtypes = []
+        self.libwarpx.warpx_PushParticlesandDepose.argtypes = [c_real]
+        self.libwarpx.warpx_getProbLo.argtypes = [ctypes.c_int]
+        self.libwarpx.warpx_getProbHi.argtypes = [ctypes.c_int]
+        self.libwarpx.warpx_getCellSize.argtypes = [ctypes.c_int, ctypes.c_int]
+        self.libwarpx.warpx_getistep.argtypes = [ctypes.c_int]
+        self.libwarpx.warpx_setistep.argtypes = [ctypes.c_int, ctypes.c_int]
+        self.libwarpx.warpx_gett_new.argtypes = [ctypes.c_int]
+        self.libwarpx.warpx_sett_new.argtypes = [ctypes.c_int, c_real]
+        self.libwarpx.warpx_getdt.argtypes = [ctypes.c_int]
+
+    def get_boundary_number(self, boundary):
```

Comment:
Previously `_get_boundary_number` was marked as a private method by the leading underscore. Should that be kept here?

Reply:
Fixed